bridges/snowbridge/pallets/inbound-queue-v2/src/lib.rs

```diff
@@ -215,9 +166,9 @@ pub mod pallet {
 		}
 	}
 
-	/// The current nonce for each channel
+	/// The nonce of the message been processed or not
 	#[pallet::storage]
-	pub type Nonce<T: Config> = StorageMap<_, Twox64Concat, ChannelId, u64, ValueQuery>;
+	pub type Nonce<T: Config> = StorageMap<_, Identity, u64, bool, ValueQuery>;
```

Comment:
The `Identity` hasher isn't safe in this context. See https://docs.substrate.io/build/runtime-storage/#hashing-algorithms for reasons:

```suggestion
	pub type Nonce<T: Config> = StorageMap<_, Twox64Concat, u64, bool, ValueQuery>;
```

Reply:
@vgeddes on the topic of nonces, how would you feel about changing this storage item to only keep pending nonces, like here: https://github.com/yrong/polkadot-sdk/pull/4/files#diff-10cb33bd6056f3960af963e776896e440e942580843f1676c6d0be393ad61010R220 When a message is processed successfully, it is removed from the map?

This approach has 2 benefits:
- The storage map will not grow boundlessly as the nonce grows
- It is easier for relayers to query this map to see which messages should be relayed, instead of querying the whole map of all previous nonces relayed and picking out the outstanding ones.

Reply:
Nevermind, brain fart. The pallet needs to keep track of which nonces have been processed otherwise a message can be processed twice. 😅

Reply:
The problem is that only the Ethereum gateway contract knows which nonces are pending.

Actually I remember now that I originally wanted to use a sparse bitmap structure to make this nonce tracking more space efficient

ie.

```
pub type Nonce<T: Config> = StorageMap<_, Twox64Concat, u128, u128, ValueQuery>
```

Each "bucket" can track 128 sequential nonces.

I do this for the Ethereum side, where the delivered inbound nonces are tracked in this data structure: https://github.com/Snowfork/snowbridge/blob/553b0b9d9ec6f3683c6c605417ea5532282771ed/contracts/src/utils/SparseBitmap.sol

Reply:
Added. 

SparseBitmap: https://github.com/claravanstaden/polkadot-sdk/pull/4/files#diff-4ff1da0a1e9f3d7498050866dabf1fe730bd941380812d0e741ef5ba7fc031d3
Updated Nonce storage: https://github.com/claravanstaden/polkadot-sdk/pull/4/files#diff-bfe761b371a553619edf5be2dd792c45bd14f1a916e51b77f5b77b9395d92cdaR174